packages/bierzo-wallet/src/routes/registerName/index.tsx

```diff
@@ -31,8 +32,25 @@ const validate = async (values: object): Promise<object> => {
     return errors;
   }
 
-  if (!username.endsWith("*iov")) {
-    errors[REGISTER_USERNAME_FIELD] = "Personalized address must include namespace suffix";
+  const checkResult = isValidIov(username);
+
+  switch (checkResult) {
+    case "not_iov":
+      errors[REGISTER_USERNAME_FIELD] = "Personalized address must include namespace suffix";
+      break;
+    case "too_short":
+      errors[REGISTER_USERNAME_FIELD] = "Personalized address should be at least 3 characters";
+      break;
+    case "too_long":
+      errors[REGISTER_USERNAME_FIELD] = "Personalized address should be maximum 64 characters";
+      break;
+    case "wrong_chars":
+      errors[REGISTER_USERNAME_FIELD] =
+        "Personalized address should contain 'abcdefghijklmnopqrstuvwxyz0123456789-_.' characters only";
+      break;
+  }
+
+  if (checkResult !== "valid") {
     return errors;
```

Comment:
The type system assumes, an `errors` object is always returned (see `Promise<object>`). So I think the condition should be removed. In case of success, `errors` is `{}`, like before

Reply:
I used this condition to avoid running validation against blockchain in next lines, this will help to save resources a little instead of validating 100% incorrect personalized address in blockchain.

Reply:
Ah, yes, perfect. I did not see the rest of the code in this function. Makes sense, my faultl